File: Minesweeper.java
import java.util.ArrayList;
import java.util.List;

public class Minesweeper {
    
    static class Bomb {
        int row;
        int col;

        Bomb(int row, int col) {
            this.row = row;
            this.col = col;
        }
    }
    
    private static int[][] findMines(List<Bomb> bombs, int numRows, int numCols) {
        int[][] field = new int[numRows][numCols];
        
        for (Bomb bomb : bombs) {
            int bomb_row = bomb.row;
            int bomb_col = bomb.col;
            
            field[bomb_row][bomb_col] = -1;
            
            int row_begin = bomb_row - 1;
            int row_end = bomb_row + 2;
            int col_begin = bomb_col - 1;
            int col_end = bomb_col + 2;
             
            for (int i = row_begin; i < row_end; ++i) {
                for (int j = col_begin; j < col_end; j++) {
                    if (0 <= i && i <= numRows && 0 <= j && j <= numCols && field[i][j] != -1) {
                        field[i][j]++;
                    }
                }            
            }
        }
        
        return field;        
    }
    
    public static void main(String[] args) {
        List<Bomb> bombs = new ArrayList<>();
        bombs.add(new Bomb(0, 0));
        bombs.add(new Bomb(1, 2));
        
        int[][] field = findMines(bombs, 3, 4);
        
        for (int row = 0; row < field.length; row++) {
            for (int col = 0; col < field[row].length; col++) {
                System.out.print(field[row][col] + " ");
            }
            System.out.println();
        }
    }
    
}

File: minesweeper.py
"""
Write a function tha will take 3 arguments:
bombs = list of bomb locations
rows, columns
mine_sweeper([0, 0], [1, 2], 3, 4)

bomb at row index 0 column index 0
bomb at row index 0 column index 1
3 rows
3 columns

We should return a 3x4 array, (-1) = bomb
"""

def minesweeper(bombs, num_rows, num_cols):
    field = [[0 for i in range(num_cols)] for j in range(num_rows)]
    
    print(field)    
    
    for bomb_location in bombs:
        (bomb_row, bomb_col) = bomb_location;
        field[bomb_row][bomb_col] = -1
    
        row_range = range(bomb_row - 1, bomb_row + 2)
        col_range = range(bomb_col - 1, bomb_col + 2)
        
        for i in row_range:
            for j in col_range:
                if (0 <= i < num_rows and 0 <= j < num_cols and field[i][j] != -1):
                    field[i][j] += 1 
            
    return field


print(minesweeper([[0, 0], [1, 2]], 3, 4))
